core/lib/dal/src/base_token_dal.rs

```diff
@@ -0,0 +1,56 @@
+use bigdecimal::BigDecimal;
+use zksync_db_connection::{connection::Connection, error::DalResult, instrument::InstrumentExt};
+
+use crate::{models::storage_base_token_price::StorageBaseTokenPrice, Core};
+
+#[derive(Debug)]
+pub struct BaseTokenDal<'a, 'c> {
+    pub(crate) storage: &'a mut Connection<'c, Core>,
+}
+
+impl BaseTokenDal<'_, '_> {
+    pub async fn insert_token_price(
+        &mut self,
+        numerator: &BigDecimal,
+        denominator: &BigDecimal,
+        ratio_timestamp: &chrono::NaiveDateTime,
+    ) -> DalResult<()> {
+        sqlx::query!(
+            r#"
+            INSERT INTO
+                base_token_prices (numerator, denominator, ratio_timestamp)
```

Comment:
there are few other values that require to be not null and without default values if you refer to the schema such as created_at or updated_at